jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/AsyncContextTest.java

```diff
@@ -224,48 +263,86 @@ public void testDispatchAsyncContext() throws Exception
     @Test
     public void testDispatchAsyncContextEncodedUrl() throws Exception
     {
-        String request = "GET /ctx/test/hello%20there?dispatch=true HTTP/1.1\r\n" +
-            "Host: localhost\r\n" +
-            "Content-Type: application/x-www-form-urlencoded\r\n" +
-            "Connection: close\r\n" +
-            "\r\n";
+        startServer((context) ->
+        {
+            ServletHolder encodedTestHolder = new ServletHolder(new TestServlet());
+            encodedTestHolder.setInitParameter("dispatchPath", "/test2/something%25else");
```

Comment:
Why has this test been changed from testing %2f to %25?
ideally we'd test both?

Reply:
%2F trips up the ambiguous stuff.
We test with %25 in ee10 for this same test.

Reply:
Keep in mind that this test, being disabled, wasn't running, so the ambiguous failure for using %2f wasn't triggering.
The point of the test is to ensure that the encoding passes through properly for the various APIs.
Which it still does by using %25 instead.